java/challenges/lib/src/main/java/challenges/stacksandqueues/Queue.java

```diff
@@ -0,0 +1,47 @@
+package challenges.stacksandqueues;
+
+import java.util.EmptyStackException;
+import java.util.NoSuchElementException;
+
+public class Queue {
+  public Node front;
+  public Node rear;
+
+  public Queue(){
+    this.front = null;
+    this.rear = null;
+  }
+
+  public void enqueue(Object val) {
+    Node newNode = new Node(val);
+    if (isEmpty()) {
```

Comment:
Nice if statement here, no need to check if the head == null, we already have a method to check that!